### PR TITLE
[WIP] Modules Hammer

### PR DESCRIPTION
**Pull request by arjunkathuria**
-  all code lintered nicely.
-  removed concat completely, build process now uses rollup by default.
- utils.js, input.js and other necessary files are broken into individual modules under the folders of same name in /src.
- ./hammer.js is rollup output, along with the min and map files.
-  use kebab dasherized case for file names, not camelCase.
-  bumped up versions for a few packages in package.json.
-  test harness also updated to point to the rollup build.
-  the rollup build is successful and pass all tests.

### CODE REVIEW

File: bundle.js [hunk 1]
```diff
@@ -0,0 +1,2691 @@
+(function(window, document, exportName, undefined) {
```
Comment (arschmitz) [inline bundle.js]: Don't change the name of the built file

**Comment by arschmitz** [comment]
You still need to generate the min and map files

**Comment by arschmitz** [comment]
The build is failing

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export default {
+    entry: 'src/main.js',
+    format: 'es6',
+    // moduleName: 'HAMMER',
+    plugins: [ babel({exclude: 'node_modules/**'}) ],
+    dest: 'bundle.js',
+    intro: " (function(window, document, exportName, undefined) { \n'use strict' ",
+    outro: "})(window, document, 'Hammer');"
+
+};
```
Comment (arschmitz) [inline rollup.config.js]: Missing new line at EOF

File: src/.babelrc [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["es2015-rollup"]
+}
```
Comment (arschmitz) [inline src/.babelrc]: Missing new line at end of file

File: src/hammer.js [hunk 4]
```diff
@@ -126,3 +137,5 @@ Hammer.defaults = {
         tapHighlightColor: 'rgba(0,0,0,0)'
     }
 };
+
+export {Hammer};
```
Comment (arschmitz) [inline src/hammer.js]: Missing new line at EOF

File: src/input/pointerevent.js [hunk 5]
```diff
@@ -86,3 +91,5 @@ inherit(PointerEventInput, Input, {
         }
     }
 });
+
+export {PointerEventInput}
```
Comment (arschmitz) [inline src/input/pointerevent.js]: Missing new line at EOF

File: src/input/pointerevent.js [hunk 6]
```diff
@@ -1,3 +1,8 @@
+import {INPUT_START,INPUT_END,INPUT_CANCEL,INPUT_MOVE,INPUT_TYPE_TOUCH,INPUT_TYPE_MOUSE,INPUT_TYPE_PEN,INPUT_TYPE_KINECT} from '../inputjs/inputConsts'
+import {Input} from '../inputjs/inputConstructor';
+import {inherit} from '../utilsjs/inherit';
+import {inArray} from '../utilsjs/inArray'
```
Comment (arschmitz) [inline src/input/pointerevent.js]: Missing semi colon

File: src/input/mouse.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
+import {INPUT_START,INPUT_MOVE,INPUT_END,INPUT_TYPE_MOUSE} from '../inputjs/inputConsts'
```
Comment (arschmitz) [inline src/input/mouse.js]: missing semi colon

File: src/input/pointerevent.js [hunk 8]
```diff
@@ -1,3 +1,8 @@
+import {INPUT_START,INPUT_END,INPUT_CANCEL,INPUT_MOVE,INPUT_TYPE_TOUCH,INPUT_TYPE_MOUSE,INPUT_TYPE_PEN,INPUT_TYPE_KINECT} from '../inputjs/inputConsts'
```
Comment (arschmitz) [inline src/input/pointerevent.js]: Missing semi colon

File: src/input/singletouch.js [hunk 9]
```diff
@@ -1,3 +1,8 @@
+import {INPUT_START,INPUT_MOVE,INPUT_END,INPUT_CANCEL,INPUT_TYPE_TOUCH} from '../inputjs/inputConsts'
+import {Input} from '../inputjs/inputConstructor'
+import {inherit} from '../utilsjs/inherit'
+import {toArray} from '../utilsjs/toArray'
+import {uniqueArray} from '../utilsjs/uniqueArray'
```
Comment (arschmitz) [inline src/input/singletouch.js]: Every line is missing the semi-colons

File: src/input/singletouch.js [hunk 10]
```diff
@@ -66,3 +71,5 @@ function normalizeSingleTouches(ev, type) {
 
     return [all, changed];
 }
+
+export {SingleTouchInput}
```
Comment (arschmitz) [inline src/input/singletouch.js]: missing semi colon

File: src/input/touch.js [hunk 11]
```diff
@@ -1,3 +1,10 @@
+import {INPUT_START,INPUT_MOVE,INPUT_END,INPUT_CANCEL,INPUT_TYPE_TOUCH} from '../inputjs/inputConsts'
```
Comment (arschmitz) [inline src/input/touch.js]: Missing semi-colon

File: src/input/touch.js [hunk 12]
```diff
@@ -1,3 +1,10 @@
...
+import {Input} from '../inputjs/inputConstructor'
```
Comment (arschmitz) [inline src/input/touch.js]: Missing Semi-colon

File: src/input/touch.js [hunk 13]
```diff
@@ -96,3 +103,5 @@ function getTouches(ev, type) {
         changedTargetTouches
     ];
 }
+
+export {TouchInput};
```
Comment (arschmitz) [inline src/input/touch.js]: Missing new line at EOF

File: src/input/touchmouse.js [hunk 14]
```diff
@@ -1,3 +1,10 @@
+import {Input} from '../inputjs/inputConstructor'
+import {inherit} from '../utilsjs/inherit';
+import {bindFn} from '../utilsjs/bindFn';
+import {TouchInput} from './touch';
+import {MouseInput} from './mouse';
+import {INPUT_START,INPUT_END,INPUT_CANCEL,INPUT_TYPE_TOUCH,INPUT_TYPE_MOUSE} from '../inputjs/inputConsts'
```
Comment (arschmitz) [inline src/input/touchmouse.js]: Missing semi-colon

File: src/input/touchmouse.js [hunk 15]
```diff
@@ -1,3 +1,10 @@
+import {Input} from '../inputjs/inputConstructor'
```
Comment (arschmitz) [inline src/input/touchmouse.js]: Missing semi-colon

File: src/input/touchmouse.js [hunk 16]
```diff
@@ -93,3 +100,5 @@ function isSyntheticEvent(eventData) {
     }
     return false;
 }
+
+export {TouchMouseInput}
```
Comment (arschmitz) [inline src/input/touchmouse.js]: Missing new line at EOF

**Comment by arschmitz** [comment]
Please review this make sure it passes both linting and JSCS and tests pass please. I look at it again once this is done

**Comment by runspired** [comment]
@arjunkathuria Could reopen this against the `feat/es6` branch?

**Pull request closed without merging**